src/lib/functions.ts

```diff
@@ -144,33 +154,64 @@ export async function trigger({
         !isRemoteServer(parsedAsyncAPI, msg.server)
       app.send(
         new GleeMessage({
+          request: message,
           payload: msg.payload,
           query: msg.query,
           headers: msg.headers,
           channel: msg.channel || message.channel,
           serverName: msg.server,
           broadcast: isBroadcast,
-        })
-      )
+        }))
     })
 
-    functionResult?.reply?.forEach((msg) => {
-      message.reply({
-        payload: msg.payload,
-        headers: msg.headers,
-        channel: msg.channel,
+    functionResult?.reply?.forEach((reply) => {
+      const replyMessage = createReply(reply, message, parsedAsyncAPI)
+      if (!replyMessage) return
+      const replyChannel = parsedAsyncAPI.channels().get(replyMessage.channel)
+      replyChannel.servers().forEach((server) => {
+        replyMessage.serverName = server.id()
+        app.send(
+          replyMessage
+        )
       })
     })
   } catch (err) {
     if (err.code === 'ERR_MODULE_NOT_FOUND') {
       const functionsPath = relative(GLEE_DIR, GLEE_FUNCTIONS_DIR)
-      const missingFile = relative(GLEE_FUNCTIONS_DIR, `${operationId}.js`)
+      const missingFile = relative(GLEE_FUNCTIONS_DIR, `${operation.id()}.js`)
       const missingPath = join(functionsPath, missingFile)
       logWarningMessage(`Missing function file ${missingPath}.`, {
         highlightedWords: [missingPath],
       })
     } else {
-      throw err
+      logError(err)
+      return
     }
   }
+}
+
+function createReply(functionReply: GleeFunctionReturnReply, message: GleeMessage, parsedAsyncAPI: AsyncAPIDocumentInterface): GleeMessage {
+  const operation = message.operation
+  const reply = operation.reply()
+  if (!reply) {
+    const warningMsg = `Operation ${operation.id()} doesn't have a reply field. the return result from your function will be ignored.`
```

Comment:
```suggestion
    const warningMsg = `Operation ${operation.id()} doesn't have a reply field. The returned result from your function will be ignored.`
```

Reply:
Is it true though? I don't think the returned result of the function will be ignored. I mean, we still process `send` if it's there, right?

Reply:
`createReply` will only be called from `reply` not `send`.
I am pretty sure it is going to get ignored but if it should? that's another discussion that we can have. :)
btw I am in favour of ignoring it since I think glee should follow spec as much as possible. no reply in spec? no reply in glee. :|